### DISCUSSION TITLE
Fix for TCH001 aka import under TYPE_CHECKING

### DISCUSSION THREAD

**Discussion by ssbarnea** [astral-sh/ruff#4252] [Ideas]
While TCH001 rule is useful one, it is also very common and it would be awesome if ruff could just fix it and move the import from top level to under `if TYPE_CHECKING:` block.

**Comment by tjkuson** [comment]
Hi! This was recently added in [ v0.0.271](https://github.com/astral-sh/ruff/releases/tag/v0.0.271).

**Comment by charliermarsh** [comment]
Ah, good call :)
